app/javascript/packages/document-capture/components/in-person-prepare-step.tsx

```diff
@@ -6,22 +6,62 @@ import {
   PageHeading,
   ProcessList,
   ProcessListItem,
+  SpinnerDots,
 } from '@18f/identity-components';
 import { removeUnloadProtection } from '@18f/identity-url';
-import { useContext } from 'react';
+import { useContext, useEffect, useState } from 'react';
 import { FlowContext } from '@18f/identity-verify-flow';
 import { useI18n } from '@18f/identity-react-i18n';
 import InPersonTroubleshootingOptions from './in-person-troubleshooting-options';
 
+const fetchSelectedLocation = () =>
+  fetch('/verify/in_person/usps_locations/selected').then((response) =>
+    response.json().catch((error) => {
+      throw error;
+    }),
+  );
+
 function InPersonPrepareStep() {
   const { t } = useI18n();
   const { inPersonURL } = useContext(FlowContext);
+  const [selectedLocationName, setSelectedLocationName] = useState<string>('');
+  const [hasFetchError, setHasFetchError] = useState<boolean>(false);
 
-  return (
-    <>
+  useEffect(() => {
+    let cancelled = false;
+    (async () => {
+      const fetchedLocation = await fetchSelectedLocation().catch((error) => {
```

Comment:
(Non-blocking) Since we had the data already on the previous screen and we know which option the user interacted with, it seems unexpected to me that we'd need to call out again to the API to get the selected location, vs. persisting that client-side between the two screens.

Reply:
we did have some discussion around retrieving the user's selected location. We could have used local storage to save and retrieve the choice, it was a much simpler solution but would introduce a security risk. The selection did need to also be saved to the session so we decided to use that route but open to hearing other possible solutions.

Reply:
These step implementations can push data into the `FormSteps` value, which is carried over between individual steps, and is only persisted within the current page session, not using storage like localStorage.

Example pseudo-code:

```tsx
function InPersonLocationStep({ onChange }) {
  function handleLocationSelect() {
    onChange({ selectedLocation: locationData[id] });
  }
}

function InPersonPrepareStep({ value }) {
  const { selectedLocation } = value;

  // ...
}
```

https://github.com/18F/identity-idp/blob/b25790b34d3c542f00849808dc37e8d0a7ec64e7/app/javascript/packages/form-steps/form-steps.tsx#L40-L54

Reply:
thank you for this suggestion! i'm going to explore this on a refactor

Reply:
implemented in #6666